httpx/_models.py

```diff
@@ -67,8 +68,8 @@
 
 HeaderTypes = typing.Union[
     "Headers",
-    typing.Dict[typing.AnyStr, typing.AnyStr],
-    typing.List[typing.Tuple[typing.AnyStr, typing.AnyStr]],
+    typing.Dict[StrOrBytes, StrOrBytes],
+    typing.Sequence[typing.Tuple[StrOrBytes, StrOrBytes]],
```

Comment:
Note: this change from `List` to `Sequence` was required, or mypy would raise errors here about `List` being invariant (so not working well with unions of types):

https://github.com/encode/httpx/blob/21f533774f87292c2381376eb40f3216a48eee06/httpx/_models.py#L492